### DISCUSSION TITLE
Providing a Translation Environment.

### DISCUSSION THREAD

**Discussion by reindex-ot** [bluesky-social/atproto#2832] [Bluesky Client App]
We would like to use Crowdin to solve the current situation where there are a large number of pull requests regarding app translation.

Crowdin's CEO has personally sent a cooperative message saying that if you use the platform, they will also do their best.

All users who do translations have expressed a desire to use Crowdin.

**Localization issues**
https://github.com/bluesky-social/social-app/issues/2940

**Crowdin CEO comment**
https://github.com/bluesky-social/social-app/issues/2940#issuecomment-2365262211

english skill a little. sorry.

**Comment by reindex-ot** [comment]
Bluesky post
https://bsky.app/profile/reindex.bsky.social/post/3l4zplsf6uz2v

**Comment by akerbeltz** [comment]
Very much needed, at the moment, people have to grab the .po and work locally and that's just prone to some many issues, not least of all duplication. Come on Bluesky, this is pretty critical...

**Comment by GaeaKat** [comment]
Crowdin would help a lot, and it can easilly it looks like export to a po file into the repo.

**Comment by smileyhead** [comment]
I would definitely like to see an improvement to the translation situation.

While I do recognise that minority issues tend to not get as much attention as ones that affect more users in general, and that localisation is by definition a minority feature, I would argue that non-English app languages are still quite important, as they are used literally 100% of the time while interacting with the app by these minorities.

The current situation sees translators not get notified about new strings, and even if they did, they cannot push updated translations by themselves and must rely on someone with write-access to do the merge for them, which can even take a week *after* the build has already been pushed to prod, leaving new functions (and potentially important one-time messages) untranslated for non-English users.

We need some sort of translator hierarchy, where a proven translator can get ‘promoted’ to a language manager of sorts, and not only be able to push updates to the localisation by themselves, but also not have to wait to be proofread by someone else. With a system like this, one language manager could oversee a localisation, approve other/newer translators' work within their language, and provide a way smoother experience for users.

Crowdin already has a system like this, where language coordinators and proofreaders are able to ‘approve’ (finalise) translations, which will always take priority in getting into the production build of the app, as opposed to non-verified submissions to a string. But even having a ‘localised’ approach here on GitHub for laxing requirements for proven translators would be nice.

**Comment by reindex-ot** [comment]
Currently, there is a backlog of localization pull requests, creating a very chaotic situation. The source for translation will not be up-to-date unless you obtain it yourself. Crowdin, which allows for the latest status and smooth management and comments, is more comfortable than exchanging comments on GitHub, and progress can be visually confirmed. If there is an environment for localization, pull requests related to language will be eliminated, making repository management easier.

**Comment by surfdude29** [comment]
A hopeful update on this: @/pfrazee has [said](https://github.com/bluesky-social/social-app/pull/6186#issuecomment-2486664738) that he's hoping to get Crowdin set up sometime soon 🤞

> It's probably time to formalize this process and get better contact with our translators! ... I think we do need to get crowdin going so that we can formalize this. I'm sorry this won't be as fast as yall deserve. We're putting out a lot of fires! But I've got this high on my task list.

**Reply by reindex-ot** [reply]
If they are willing to make a move, I will patiently wait.

**Comment by surfdude29** [comment]
Great news! Bluesky is now using Crowdin! Please see https://github.com/bluesky-social/social-app/issues/7685 for all the details.
